binding/Binding/SKDocument.cs

```diff
@@ -3,25 +3,19 @@
 
 namespace SkiaSharp
 {
-	public class SKDocument : SKObject
+	public class SKDocument : SKObject, ISKReferenceCounted
 	{
 		public const float DefaultRasterDpi = 72.0f;
 
+		// keep the stream alive for as long as the document exists
+		private SKWStream underlyingStream;
```

Comment:
The streams were in danger of being disposed while the document was in progress. To fix this, we keep a reference to it to make sure the GC doesn't pick it up early.